metabase/metabase
# Search Milestone 1.5 - Integration Branch

_Note: the code in this PR has been reviewed in their separate PR branches. We just need to make sure it all works properly according to the PDs/PMs._

### Description

Encapsulates all of the branches for the initial part of the search sidebar refactor, which includes the new type filter, a new CTA at the bottom of the search dropdown, and the removal of the old search filter modal. Each PR also includes unit and E2E tests.

The tasks included in this integration branch are:
- [x] [Remove and rename SearchFilter files for Milestone 1.5 | metabase#33532](https://github.com/metabase/metabase/pull/33532)
- [x] [Add Type Filter to Search | metabase#33639](https://github.com/metabase/metabase/pull/33639)
- [x] [Add CTA to search dropdown | metabase#33675](https://github.com/metabase/metabase/pull/33675)
- [x] [Convert SearchResults.jsx to typescript | metabase#33712](https://github.com/metabase/metabase/pull/33712)
- [x] [FE] Remove Metrics / Segments from list of types
- [x] [FE] Reorder types list to Dashboard / Question / Model / Collection / Database / Table

### How to verify
- All tests should pass
- Search should be working according to the new specs laid out in the [Notion doc](https://www.notion.so/metabase/Improve-filtering-in-search-results-e7d81105e9ad4443bad3a05670be8563?pvs=4)


### Checklist
